src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java

```diff
@@ -84,17 +84,36 @@ public static final int compareLocatables(Locatable first, Locatable second, SAM
             // compare the contigs
             result = compareContigs(first, second, dictionary);
             if (result == 0) {
-                // compare start position
-                result = Integer.compare(first.getStart(), second.getStart());
-                if (result == 0) {
-                    // compare end position
-                    result = Integer.compare(first.getEnd(), second.getEnd());
-                }
+                result = compareLocatablesOfSameContig(first, second);
```

Comment:
This change checks again tha the contigs are the same, which was already done in `compareContigs`. I think that this produces a performance regression...

Reply:
Thanks, and I see your concern.
The reason I did that check in the new refactored out method, which is `public`, is to prevent callers from accidentally using it for locatables from different contigs, so a safe-guard.
Do you have better suggestions?

Reply:
What's about a private method `comparePosition` without the check, which is called by both `compareLocatables` and `compareLocatablesOfSameContig`? Like that, there is no code repetition (less error prone) and no overhead of checking twice in the methods that does not need it.